src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs

```diff
@@ -173,7 +173,14 @@ protected override void Dispose(bool disposing)
                         var attr = File.GetAttributes(copiedFile);
                         if (!attr.HasFlag(FileAttributes.Directory))
                         {
-                            File.SetLastWriteTimeUtc(copiedFile, entry.LastWriteTime.UtcDateTime);
+                            try
+                            {
+                                File.SetLastWriteTimeUtc(copiedFile, entry.LastWriteTime.UtcDateTime);
+                            }
+                            catch (ArgumentOutOfRangeException)
```

Comment:
Is it possible to check the value of `LastWriteTime` and replace `try-catch` block with `if-then`? Less expensive?